### ISSUE TITLE
I'm really confused with the v2.0 meta.py

### ISSUE THREAD

**Issue by yaox12** [dragen1860/MAML-Pytorch#14]
- First, the comment says the index of `losses_q` is tasks index.
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L77
However, in each task i , the whole list is updated.
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L94
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L105
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L123

- Second, I haven't seen the sum of loss_q?
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L134-L135
`losses_q[-1]` seems to be the last step's loss for the last task?

- Third, if `update_step == 1`, there will be only one inner update. However, the loss after first update is computed under `torch.no_grad()`, so I think there is no backward update information on the query set.
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L100-L109

**Comment by dragen1860** [comment]
1.  index of `losses_q` is NOT tasks index, but the update step index. sry for wrong comments.

2. sum of `loss_q` is accumulated at every inner update, hence just need to average it.
losses_q[-1] is the last step's loss.

3.yes, you are right. For only single step's update setting, the code `loss_q = F.cross_entropy(logits_q, y_qry[i]) ` should be moved out of `torch.no_grad()`.

**Comment by yaox12** [comment]
At the end of each task i, you append `loss_q` to list `losess_q`
https://github.com/dragen1860/MAML-Pytorch/blob/fc20b31e3927bae245376a4e46f45ea20363c30c/meta.py#L130-L131
The accumulated last step's loss should be `losses_q[self.update_step]` instead of `losses_q[-1]`, because the length of `losses_q` is `update_step + 1 + task_num` in the end.
In fact, I think the above two lines are redundant and useless.
